dandischema/models.py

```diff
@@ -1580,6 +1580,20 @@ class CommonModel(DandiBaseModel):
     wasGeneratedBy: Optional[Sequence[Activity]] = Field(
         None, json_schema_extra={"nskey": "prov"}
     )
+
+    # Our current draft dandisets could be coming from Published ones,
+    # and thus carry their attributes. So we would need to define them here as well
+    # but make them optional, and `Publishable` would overload to make
+    # them mandatory.
```

Comment:
when published, the draft state should remove the published by component. or we need a better flow that allows owners to create a new draft state from the latest published record (instead of defaulting to creating one). originally the process mimicked github in that a release is a tag on a branch. we can evolve that to be a staged evolution. draft/unpublished -> published -> create draft/unpublished when any change is made by the owner. i.e. any modification creates a new staged state. when that happens irrelevant attributes can be removed. 

but it can continue to have `derivedFrom` to indicate that this new unpublished state did come from a previous published state.